pyccel/stdlib/cwrapper/cwrapper.c

```diff
@@ -0,0 +1,600 @@
+}
+//-----------------------------------------------------//
+PyObject	*Bool_to_PyBool(bool *b)
+{
+	return *b == true ? Py_True : Py_False;
+}
+//-----------------------------------------------------//
+PyObject	*Int64_to_PyLong(int64_t *i)
+{
+	PyObject	*object;
+
+	object = PyLong_FromLongLong((long long) *i);
+
+	return object;
+}
+//-----------------------------------------------------//
+PyObject	*Int32_to_PyLong(int32_t *i)
+{
+	PyObject	*object;
+
+	object = PyLong_FromLongLong((long long) *i);
+
+	return object;
+}
+//-----------------------------------------------------//
+PyObject	*Int16_to_PyLong(int16_t *i)
+{
+	PyObject	*object;
+
+	object = PyLong_FromLongLong((long long) *i);
+
+	return object;
+}
+//--------------------------------------------------------//
+PyObject	*Int8_to_PyLong(int8_t *i)
+{
+	PyObject	*object;
+
+	object = PyLong_FromLongLong((long long) *i);
+
+	return object;
+}
+//--------------------------------------------------------//
+PyObject	*Double_to_PyDouble(double *d)
+{
+	PyObject	*object;
+
+	object = PyFloat_FromDouble(*d);
+
+	return object;
+}
+//--------------------------------------------------------//
+PyObject	*Float_to_PyDouble(float *d)
+{
+	PyObject	*object;
+
+	object = PyFloat_FromDouble((double)*d);
+
+	return object;
+}
+
+/*
+ * Functions : Cast functions
+ * ---------------------------
+ * Some of the function used below are based on C/python api and numpy/c api with
+ * more tolerance to different precisions, different system architectures and complex type.
+ * Check the C data type ob a python object
+ * Parameters :
+ *     object     : the python object
+ *     hard_check : boolean true if intensive precision check is needed
+ *
+ * Returns    :
+ *     boolean : logic statement responsible for checking python data type
+ * Reference of the used c/python api function
+ * ---------------------------------------------------
+ * https://docs.python.org/3/c-api/long.html#c.PyLong_Check
+ * https://docs.python.org/3/c-api/complex.html#c.PyComplex_Check
+ * https://docs.python.org/3/c-api/float.html#c.PyFloat_Check
+ * https://docs.python.org/3/c-api/bool.html#c.PyBool_Check
+ * https://numpy.org/doc/1.17/reference/c-api.array.html#c.PyArray_IsScalar
+ */
+
+bool    PyIs_Int8(PyObject *o, bool hard_check)
+{
+	if (hard_check == true)
+		return PyArray_IsScalar(o, Int8);
+
+	return PyLong_Check(o) || PyArray_IsScalar(o, Int8);
+}
+//--------------------------------------------------------//
+bool    PyIs_Int16(PyObject *o, bool hard_check)
+{	
+	if (hard_check == true)
+		return PyArray_IsScalar(o, Int16);
+
+	return PyLong_Check(o) || PyArray_IsScalar(o, Int16);
+}
+//--------------------------------------------------------//
+bool    PyIs_Int32(PyObject *o, bool hard_check)
+{
+	#ifdef _WIN32
+		return PyLong_Check(o) || PyArray_IsScalar(o, Int32);
+	#endif
+
+	if (hard_check == true)
+		return PyArray_IsScalar(o, Int32);
+
+	return PyLong_Check(o) || PyArray_IsScalar(o, Int32);
```

Comment:
You need to use `#ifdef`, `#else`, `#endif` to avoid warnings about unreachable code. Test that pass the `-Werror` flag cause these warnings to make the test fail.

Reply:
Fixed in 7661749